tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Logging/ILogger/LogsInjection/DatadogLoggingScope.cs

```diff
@@ -43,39 +48,36 @@ internal DatadogLoggingScope(Tracer tracer)
         {
             get
             {
-                // For mismatch version support we need to keep requesting old keys.
-                var distContext = _tracer.DistributedSpanContext;
                 return index switch
                 {
                     0 => new KeyValuePair<string, object>("dd_service", _service),
                     1 => new KeyValuePair<string, object>("dd_env", _env),
                     2 => new KeyValuePair<string, object>("dd_version", _version),
-                    3 => new KeyValuePair<string, object>("dd_trace_id", distContext?[SpanContext.Keys.TraceId] ?? distContext?[HttpHeaderNames.TraceId] ?? "0"),
-                    4 => new KeyValuePair<string, object>("dd_span_id", distContext?[SpanContext.Keys.ParentId] ?? distContext?[HttpHeaderNames.ParentId] ?? "0"),
+
+                    3 => new KeyValuePair<string, object>(
+                        "dd_trace_id",
+                        _tracer.DistributedSpanContext is { } context && LogContext.TryGetTraceId(context, _use128Bits, out var traceId) ? traceId : "0"),
+
+                    4 => new KeyValuePair<string, object>(
+                        "dd_span_id",
+                        _tracer.DistributedSpanContext is { } context && LogContext.TryGetSpanId(context, out var spanId) ? spanId : "0"),
```

Comment:
This is the only case where we want to get trace id and span id separately, hence the `TryGetTraceId()` and `TryGetSpanId()` methods.